src/m365/pp/commands/solution/solution-publisher-add.spec.ts

```diff
@@ -0,0 +1,152 @@
+  //#endregion
+
+  let log: string[];
+  let logger: Logger;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post,
+      powerPlatform.getDynamicsInstanceApiUrl
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.SOLUTION_PUBLISHER_ADD), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['publisherid', 'uniquename', 'friendlyname']);
+  });
+
+  it('fails validation if choiceValuePrefix contains invalid format.', async () => {
+    const actual = await command.validate({
+      options: {
+        environment: validEnvironment,
+        name: validName,
+        displayName: validDisplayName,
+        prefix: validPrefix,
+        choiceValuePrefix: 'Not A Number'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, name: validName, displayName: validDisplayName, prefix: validPrefix, choiceValuePrefix: validChoiceValuePrefix } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('correctly adds a specific publisher with the required parameters', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').callsFake(async () => envUrl);
+
+    sinon.stub(request, 'post').callsFake(async opts => {
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.0/publishers`)) {
+        if ((opts.headers?.accept as string)?.indexOf('application/json') === 0) {
+          return;
+        }
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { verbose: true, environment: validEnvironment, name: validName, displayName: validDisplayName, prefix: validPrefix, choiceValuePrefix: validChoiceValuePrefix } });
+    assert(loggerLogToStderrSpy.called);
```

Comment:
I think we can safely remove this loggerspy. This test should not test that we log verbose messages.